tests/test_mig_shared_transferfunctions.py

```diff
@@ -0,0 +1,208 @@
+# -*- coding: utf-8 -*-
+#
+# --- BEGIN_HEADER ---
+#
+# test_mig_shared_transferfunctions - unit test of the corresponding mig shared module
+# Copyright (C) 2003-2024  The MiG Project by the Science HPC Center at UCPH
+#
+# This file is part of MiG.
+#
+# MiG is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# MiG is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+# USA.
+#
+# --- END_HEADER ---
+#
+
+"""Unit tests for the migrid module pointed to in the filename"""
+
+import os
+import sys
+import tempfile
+
+sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__))))
+
+from tests.support import TEST_OUTPUT_DIR, MigTestCase, FakeConfiguration, \
+    cleanpath, temppath, testmain
+from mig.shared.transferfunctions import get_transfers_path, \
+    load_data_transfers, create_data_transfer, delete_data_transfer, \
+    lock_data_transfers, unlock_data_transfers
+
+DUMMY_USER = "dummy-user"
+DUMMY_ID = "dummy-id"
+DUMMY_HOME_DIR = 'dummy_user_home'
+DUMMY_HOME_PATH = os.path.join(TEST_OUTPUT_DIR, DUMMY_HOME_DIR)
+DUMMY_SETTINGS_DIR = 'dummy_user_settings'
+DUMMY_SETTINGS_PATH = os.path.join(TEST_OUTPUT_DIR, DUMMY_SETTINGS_DIR)
+
+
+def noop(*args, **kwargs):
+    if args:
+        return args[0]
+    return None
+
+
+class MigSharedTransferfunctions(MigTestCase):
+    """Wrap unit tests for the corresponding module"""
+
+    def test_transfers_basic_locking(self):
+        os.makedirs(os.path.join(DUMMY_HOME_PATH, DUMMY_USER))
+        cleanpath(DUMMY_HOME_DIR, self)
+        os.makedirs(os.path.join(DUMMY_SETTINGS_PATH, DUMMY_USER))
+        cleanpath(DUMMY_SETTINGS_DIR, self)
+        dummy_conf = FakeConfiguration(user_home=DUMMY_HOME_PATH,
+                                       user_settings=DUMMY_SETTINGS_PATH)
+        transfers_path = get_transfers_path(dummy_conf, DUMMY_USER)
+
+        # Lock shared twice should be fine
+        ro_lock = lock_data_transfers(transfers_path, exclusive=False)
+        ro_lock_again = lock_data_transfers(transfers_path, exclusive=False)
+        assert(ro_lock and ro_lock_again)
```

Comment:
Assertions within the tests should be using the TestCase assertion methods as opposed to assert statements - wile they both raise AssertionError exception objects (an extremely useful property of the design of the unittest library) an `assert` communicates an invariant within the program while the assertion methods are purposefully named much more descriptively and communicate something specific about the values being checked. Hence, e.g. `self.assertTrue()`, self.assertEqual()`, etc.

Second, in this specific case - these assertions are and'ing two _separate_ ideas together - we'd like to establish that the lock was a acquired and the _reacquisition_ of the lock is a separate thing, that might succeed or fail separately. Thus, these would be better written as two separate assertions, first self.assertTrue(ro_lock) then try to lock again and a separate self.assertTrue(ro_lock_again). Note, the larger idea is that a test failure would hint to exactly what failed, and in this case it would be far more "straight to the cause" to know whether we broke the initial lock or the relock.